res/controllers/Pioneer-DDJ-200-scripts.js

```diff
@@ -1,406 +1,594 @@
+            if (value) {
+                if (engine.getValue("[Skin]", "show_4decks")) {
+                    theDeck.toggle();
+                    theDeck.padUnit.updateDeck(theDeck.currentDeck);
+                    if (theDeck.currentDeck === "[Channel3]" || theDeck.currentDeck === "[Channel4]") {
+                        this.blinkConnection = engine.makeConnection("[App]", "indicator_250ms", function(value, _group, _control) {
+                            theDeck.syncButton.send(theDeck.syncButton.on * value);
+                        });
+                    } else if (this.blinkConnection) {
+                        this.blinkConnection.disconnect();
+                    };
+                } else {
+                    const currentRange = engine.getValue(this.group, "rateRange");
+                    engine.setValue(this.group, "rateRange", (currentRange>0.9)?0.1:currentRange+0.1);
+                };
+            };
+        },
+    });
+
+    this.playButton = new components.PlayButton({
+        midi: [0x8F + midiChannel, 0x0B],
+        shiftChannel: false,
+        shiftControl: true,
+        shiftOffset: 0x3C,
+        inputFaderStart: function(channel, control, value, status, _g) {
+            this.inKey = "play";
+            this.input(channel, control, value, status, _g);
+        },
+    });
+
+    this.cueButton = new components.CueButton({
+        midi: [0x8F + midiChannel, 0x0C],
+        shiftChannel: false,
+        shiftControl: true,
+        shiftOffset: 0x3C,
+        outKey: "cue_indicator",
+        inputFaderStop: function(channel, control, value, status, _g) {
+            this.inKey = "cue_set";
+            this.input(channel, control, value, status, _g);
+            this.inKey = "cue_gotoandstop";
+            this.input(channel, control, value, status, _g);
+        },
+    });
+
+    this.pflButton = new components.Button({
+        midi: [0x8F + midiChannel, 0x54],
+        key: "pfl",
+        type: components.Button.prototype.types.toggle,
+    });
+
+    this.loadTrack = new components.Button({
+        midi: [0x8F + midiChannel, 0x68],
+        inKey: "LoadSelectedTrack",
+    });
+
+    this.volume = new components.Pot({
+        inKey: "volume",
+    });
+
+    this.rate = new components.Pot({
+        inKey: "rate",
+        invert: 1,
+    });
+
+    this.eqKnob = [];
+    for (let i = 0; i < 3; i++) {
+        this.eqKnob[i] = new components.Pot({
+            group: `[EqualizerRack1_${this.currentDeck}_Effect1]`,
+            inKey: `parameter${i+1}`,
+        });
+    };
 
-DDJ200.switchPadLED = function(deck, pad, enabled) {
-    midi.sendShortMsg(0x97 + 2 * deck, pad - 1, 0x7F * enabled);
-};
+    this.super1 = new components.Pot({
+        group: (engine.getSetting("useSuperAsGain"))?theDeck.currentDeck:`[QuickEffectRack1_${this.currentDeck}]`,
+        inKey: (engine.getSetting("useSuperAsGain"))?"pregain":"super1",
+    });
```

Comment:
NIT (this is easier to read):

```suggestion
    if (engine.getSetting("useSuperAsGain")) {
        this.super1 = new components.Pot({
            group: theDeck.currentDeck,
            inKey: "pregain",
        });
    } else {
        this.super1 = new components.Pot({
            group: `[QuickEffectRack1_${this.currentDeck}]`;
            inKey: "super1",
        });
    }
```